addons/stock/stock.py

```diff
@@ -4232,10 +4232,11 @@ def _complete_name(self, cr, uid, ids, name, args, context=None):
         """
         res = {}
         for m in self.browse(cr, uid, ids, context=context):
-            res[m.id] = m.name
+            res[m.id] = m.name or ''
             parent = m.parent_id
             while parent:
-                res[m.id] = parent.name + ' / ' + res[m.id]
+                parent_name = parent.name or ''
+                res[m.id] = "%s / %s" %(parent_name, res[m.id])
```

Comment:
```res[m.id] = u"%s / %s" % (parent_name, res[m.id])```

(unicode or you may get funky decoding errors if parent_name or res[m.id] contains non ascii chars)